update RBAC documentation after introducing new roles (#1378)
mendersoftware/mender-docs: 02.Overview/12.Role.Based.Access.Control/docs.md

* **Admin:** Full access

> * **Read Access:** Allows the user to see the status of devices and deployments,

Reviewer: From what I understood Fabio, this should be in the description to:

> Unless the user is assigned Admin, this role is needed to allow access to the UI.

Author: updated

but not make any modifications. This role is well suited for limited technical
support users, or team leads who need an overview of deployment status or
individual devices, but are not involved in day-to-day deployment management.

* **Release Manager:** Intended for *Continuous Integration* systems. It can only manage
Mender Artifacts, such as upload and delete Artifacts.

* **Deployments Manager:** Intended for users responsible for managing deployments.
With this role users can create and abort deployments.

* **Troubleshooting:** User with this role assigned has access
to the troubleshooting features such as Remote Terminal, File Transfer, Port Forwarding.

> * **Device groups:** Grants access to a *customizable* set of Device groups in

Reviewer: For the device groups I think it's important to add how to actually get a result where "a user can achieve full control but only over a single group".  That part isn't evident.

Would expand with:
> To have a user who has full control over a particular group and nothing else be sure to give to give them the roles: Deployments Manager, Read Access and the chosen Device group role.

Author: updated

* **Admin:** Full access

* **Read Access:** Unless the user is assigned Admin, this role is needed
to allow access to the UI. The role allows the user to see the status of devices and deployments,
but not make any modifications. This role is well suited for limited technical
support users, or team leads who need an overview of deployment status or
individual devices, but are not involved in day-to-day deployment management.

* **Release Manager:** Intended for *Continuous Integration* systems. It can only manage

> Mender Artifacts, such as upload and delete Artifacts.

Reviewer: ```suggestion
Mender Artifacts, such as upload and delete Artifacts. On its own doesn't allow UI access.
```

Author: done

* **Deployments Manager:** Intended for users responsible for managing deployments.

> With this role users can create and abort deployments.

Reviewer: ```suggestion
With this role users can create and abort deployments.  On its own doesn't allow UI access.
```

Author: done

* **Troubleshooting:** User with this role assigned has access

> to the troubleshooting features such as Remote Terminal, File Transfer, Port Forwarding.

Reviewer: ```suggestion
to the troubleshooting features such as Remote Terminal, File Transfer, Port Forwarding.  On its own doesn't allow UI access.
```

Author: done

* **Admin:** Full access

> * **Read Access:** Unless the user is assigned Admin, this role is needed
> to allow access to the UI. The role allows the user to see the status of devices and deployments,

Reviewer: ```suggestion
* **Read Access:** The role allows the user to see the status of devices and deployments,
```

Author: done

but not make any modifications. This role is well suited for limited technical
support users, or team leads who need an overview of deployment status or
individual devices, but are not involved in day-to-day deployment management.

* **Release Manager:** Intended for *Continuous Integration* systems. It can only manage

> Mender Artifacts, such as upload and delete Artifacts. On its own doesn't allow UI access.

Reviewer: ```suggestion
Mender Artifacts, such as upload and delete Artifacts.
```

Author: done

* **Deployments Manager:** Intended for users responsible for managing deployments.

> With this role users can create and abort deployments. On its own doesn't allow UI access.

Reviewer: ```suggestion
With this role users can create and abort deployments. On it's own this role won't make the devices visible in the UI, you must pair it with Read Access for that```

Author: done

* **Troubleshooting:** User with this role assigned has access
to the troubleshooting features such as Remote Terminal, File Transfer, Port Forwarding.

> On its own doesn't allow UI access.

Reviewer: ```suggestion
On it's own this role won't make the devices visible in the UI, you must pair it with Read Access for that.
```

Author: done
